server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/symbols/SymbolAccumulatorService.java

```diff
@@ -82,13 +96,32 @@ public List<Location> getDefinition() {
     public List<Location> getReferences() {
       return usage;
     }
-  };
 
-  private final Map<String, FunctionInfo> functions;
+    /**
+     * Checks if a function is declared under a passed programNode
+     * @param programNode
+     * @return true if a function is declared under a passed programNode, false otherwise
+     */
+    public boolean isDeclared(ProgramNode programNode) {
+      if (programNode == null) return false;
+      while (!this.getDeclaredProgramNode().contains(programNode)) {
```

Comment:
Does this mean that the repositories are inherited?

Reply:
Yes, in case of nested cobol program. Added a test [here](https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/014d22ed1006a92b68f1fe655fd2c4a6a0d2e634/server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestFunction.java#L555)